Add RN2483

dbrgn opened the pull request:
Datasheet: http://ww1.microchip.com/downloads/en/DeviceDoc/50002346C.pdf

Symbol:

![img](https://tmp.dbrgn.ch/screenshots/20190127020132-07_0fsel.png)

Package:

![img](https://tmp.dbrgn.ch/screenshots/20190127020150-aaylik73.png)

Device:

![img](https://tmp.dbrgn.ch/screenshots/20190127020159-h9e0pbkq.png)

dbrgn committed:
Add RN2483

rnestler commented on cmp/6b9f28de-f963-4511-b508-d8b33b4f8d60/component.lp:
```diff
@@ -0,0 +1,129 @@
+ (signal fc78dbd7-d15e-45a3-a56d-3e4481309e40 (name "PGC_INT") (role passive)
+  (required false) (negated false) (clock false) (forced_net "")
+ )
+ (signal 20d25de7-8623-4856-a0b9-fb238fa0e1b7 (name "nRESET") (role passive)
```
Do we have a naming convention for negated inputs? I couldn't find it on https://docs.librepcb.org/#libraryconventions-symbols-naming.
Is the `n` prefix the convention or is this just the name used on the datasheet?

dbrgn replied:
That's a good point, I dimly remember that @ubruhin had an opinion on that, but I just did a random choice for now. In the schematics, there's a RESET pin with an overline.

Would be nice to put that in the conventions.

rnestler commented on library.lp:
```diff
@@ -3,7 +3,7 @@
  (keywords "")
  (author "LibrePCB")
- (version "0.2.0")
+ (version "0.2")
```
What versioning scheme do we recommend to use for libraries? Semantic versioning?

dbrgn replied:
Note: This change was done automatically by the current dev version of LibrePCB.

My take would be incrementing integers instead of semver-looking version numbers. I'll explain more here: https://librepcb.discourse.group/t/library-versioning/79

rnestler commented on cmp/6b9f28de-f963-4511-b508-d8b33b4f8d60/component.lp:
```diff
@@ -0,0 +1,129 @@
+(librepcb_component 6b9f28de-f963-4511-b508-d8b33b4f8d60
+ (name "RN2483")
+ (description "Low-Power Long Range LoRa Technology Transceiver Module.")
+ (keywords "rn2483,rn2483a,lorawan,lora,transceiver")
```
What recommendations for keywords do we have? All lowercase? Is only English / ASCII recommended? Do / should we translate keywords?

dbrgn replied:
Maybe start a discussion in the discourse forum?

rnestler commented on dev/190f7529-eb16-4963-9e23-78b56f45c9e1/device.lp:
```diff
@@ -0,0 +1,59 @@
+ (version "0.1")
+ (created 2019-01-27T00:48:03Z)
+ (deprecated false)
+ (category f7f9bc34-7966-4af4-8cd4-4b8101e0823a)
```
Shouldn't this have the `4e89bf79-0506-4b24-8616-b0461a5932e0` which is created below?

dbrgn replied:
No, this is a component category (`cmpcat`), not a package category (`pkgcat`). The latter is only used for packages.

I'm currently using the cmpcat "Integrated Circuits > Interface > Drivers, Receivers, Transmitters". Hope that's correct.

rnestler reviewed (commented):
Mostly comments about conventions which weren't clear for me by reading https://docs.librepcb.org/#libraryconventions. This may be valuable because I was quite a bit out of the loop in that regard :wink:

ubruhin commented:
For me it looks a bit strange that the package is named "Microchip RN2483", while all other elements are named just "RN2483", but somehow I don't have an opinion how they should be named instead :grin:

Symbol:
- Could we maybe decrease its width by 5.08mm or would it look ugly then?

Package: 
- Why is pad 1 not at top left?
- The courtyard shouldn't be marked as "grab area".
- Maybe make the silkscreen line a bit longer at pin 1 to have a pin-1 marking (although the assembly orientation is clear anyway, a pin-1 marking for a package with that many pads might be helpful for debugging as it's easier to find a specific pad on a PCB without opening the EDA project)?

dbrgn committed:
fixup: Remove vendor prefix from pkg name

dbrgn committed:
fixup: Reduce width of sym

dbrgn committed:
fixup: Rotate pkg

dbrgn committed:
fixup: Don't make courtyard a grab area

dbrgn committed:
fixup: Pin 1 marking

dbrgn commented:
> For me it looks a bit strange that the package is named "Microchip RN2483", while all other elements are named just "RN2483"

I removed the manufacturer prefix.

> Could we maybe decrease its width by 5.08mm or would it look ugly then?

Looks fine!

![img](https://tmp.dbrgn.ch/screenshots/20190202155350-6noskm1b.png)

> Why is pad 1 not at top left?

Because I copied the package from the datasheet. But you're right that it should be rotated - fixed.

> The courtyard shouldn't be marked as "grab area".

Fixed

> Maybe make the silkscreen line a bit longer at pin 1 to have a pin-1 marking

Good idea, done:

![img](https://tmp.dbrgn.ch/screenshots/20190202155958-tiklamte.png)
